[14.0] account_reconciliation_widget: fix notifications, show analytic and add ability to edit 'ref'

alexis-via opened the pull request:
With this PR, 'ref' of account.move can now be customized in the reconciliation widget and 'name' of account.move is displayed for information.

One of my users asked me for this ; he wanted this for his classification/archive methodology. I decided to propose this development here: if you think it's useful for the module, we can review/merge it. If you think this feature is not useful for the module, I'll maintain this branch for that specific user and we cancel the PR.

![reconcile_widget-edit_ref](https://user-images.githubusercontent.com/1157917/182189688-1cfd453d-0166-44ea-89c6-5ecdb8db5ef2.png)

alexis-via committed:
account_reconciliation_widget: ability to edit ref

'ref' of account.move can now be customized in the reconciliation widget
'name' of account.move is displayed for information

pedrobaeza commented:
The screenshot is of the "after the PR"? Isn't the labels unaligned?

alexis-via commented:
Yes, the screenshot is after the PR ; as you can see, the "Reference" field appear at the top and there is the move number next to it.

The label alignement problem at the bottom of the screen is unrelated. I reduced the screen width to make the screenshot ; on a regular PC screen, the fields are spread access 2 columns, so you don't see this alignment problem.

pedrobaeza commented:
OK, I see. Not sure if this should go in the main module. Let see others what think.

alexis-via commented:
IMHO:

Pros : 

- adds a possibility for accountants

Cons:

- 67 more lines of code to maintain. But this module has 10k lines of code (5k if you exclude tests), so it's not very significant.

Remark:

- without this patch, the 'ref' field contains the 'name' of the account.bank.statement, which, in real life, doesn't contain any useful information. With this PR, the 'ref' field will be empty by default ; it will have a value only if the accountant sets the value in the JS reconciliation interface.

andreampiovesana reviewed (approved):
LGTM

alexis-via committed:
[FIX] account_reconciliation_widget: fix JS crash when there are not taxes

alexis-via committed:
[FIX] account_reconcilication_widget: notifications

Notifications displayed at the start of the reconciliation interface are
back up and running (it was working in v10, broken in v12 and v13).
For example, when you import an OFX file and go directly to the
reconciliation interface, it will display the number of transactions
that were automatically dropped because they were already present in
Odoo (detected via the unique_import_id).

alexis-via commented:
Another killer feature of this PR: notifications when starting the reconcile interface are working again (they were working on v10 and, according to my tests, broken in v12 and v13). Here is an example on v14 with this PR when importing an OFX file that had 7 transactions that were already imported in a previous OFX file:

![bank_statement_v14-notifications](https://user-images.githubusercontent.com/1157917/183762521-7eacbceb-8be1-4d78-8b05-77e01f3eb174.png)

clementmbr reviewed (approved):
LGTM! Thanks

alexis-via committed:
[IMP] account_reconciliation_widget: show analytic account on counterpart line

alexis-via commented:
I also added the display of the analytic account on the counterpart line. Before that, when you had several counter-part lines, you could not see the analytic account you had set on the previous counterpart lines.

![bank-statement-rec-show_analytic](https://user-images.githubusercontent.com/1157917/184030112-0e140ce4-f66f-487e-9320-2b873d470415.png)

I display the analytic account between the due date and the label. If you think another position would be better, please suggest and explain why you think it's better... I'm open to improvement !

More details tomorrow after a good night of sleep :)

njeudy reviewed (approved):
Great job I just tested it and think accountant will love it :) 👍

pedrobaeza commented:
/ocabot merge major

pedrobaeza commented:
Can you please fw-port it to v15?

pedrobaeza commented:
@alexis-via please fw-port it to v15
